src/Transfer.js

```diff
@@ -0,0 +1,403 @@
+import { AddIndividual } from './Transfer/AddIndividual.js'
+import { Container } from './Transfer/Container.js'
+import { Filter } from './Transfer/Filter.js'
+import { LeftFooter } from './Transfer/LeftFooter.js'
+import { LeftHeader } from './Transfer/LeftHeader.js'
+import { LeftSide } from './Transfer/LeftSide.js'
+import { PickedOptions } from './Transfer/PickedOptions.js'
+import { RemoveAll } from './Transfer/RemoveAll.js'
+import { RemoveIndividual } from './Transfer/RemoveIndividual.js'
+import { ReorderingActions } from './Transfer/ReorderingActions.js'
+import { RightFooter } from './Transfer/RightFooter.js'
+import { RightSide } from './Transfer/RightSide.js'
+import { SourceOptions } from './Transfer/SourceOptions.js'
+import {
+    addAllSelectableSourceOptions,
+    addIndividualSourceOptions,
+    createDoubleClickHandlers,
+    extractPickedReactOptions,
+    defaultFilterCallback,
+    filterOutOptions,
+    getSubsetByFilter,
+    isReorderDownDisabled,
+    isReorderUpDisabled,
+    moveHighlightedPickedOptionDown,
+    moveHighlightedPickedOptionUp,
+    removeAllPickedOptions,
+    removeIndividualPickedOptions,
+    useHighlightedOptions,
+} from './Transfer/helper/index.js'
+import {
+    singleSelectedPropType,
+    multiSelectedPropType,
+} from './common-prop-types.js'
+
+/**
+ * @module
+ * @param {Transfer.PropTypes} props
+ *
+ * @returns {React.Component}
+ *
+ * @example import { Transfer } from @dhis2/ui-core
+ * @see Specification: {@link https://github.com/dhis2/design-system/blob/master/organisms/transfer.md|Design system}
+ * @see Live demo: {@link /demo/?path=/story/transfer--basic|Storybook}
+ *
+ * This component has to differentiate between different types of options,
+ * here's an explanation of their meaning:
+ *
+ * * source options -> These are options listed on the left and are available
+ * for selection
+ *
+ * * picked options -> These options have been selected by the user and are on
+ * the right side
+ *
+ * * highlighted option -> These are visually highlighted options than can be
+ * on either side and are ready for transferral with the action buttons to the
+ * other side
+ *
+ * * filtered options -> These are the displayed source options filtered
+ * by a search term or a custom search algorithm.
+ *
+ * The api surface uses "selected" for "picked" to be consistent with the
+ * rest of the library
+ */
+export const Transfer = ({
+    onChange,
+
+    children,
+    className,
+    dataTest,
+    disabled,
+    sourceEmptyPlaceholder,
+    selectedEmptyComponent,
+    enableOrderChange,
+    filterLabel,
+    filterCallback,
+    height,
+    initialSearchTerm,
+    addAllText,
+    addIndividualText,
+    removeAllText,
+    removeIndividualText,
+    leftFooter,
+    leftHeader,
+    maxSelections,
+    optionsWidth,
+    rightFooter,
+    searchTerm,
+    selected,
+    selectedWidth,
+    showFilterInput,
+    onFilterChange,
+}) => {
+    /**
```

Comment:
Are these comments intended to show up in the JSDocs?

Leading a comment with two stars (`/**`) means JSDocs is going to pick it up. If these comments are meant for the dev reading the source code only, then we should use a single star comment (`/*`).

Reply:
Thanks. I've removed one asterix from all internal comments